workers/scanner.go

```diff
@@ -0,0 +1,151 @@
+package workers
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/skynetlabs/pinner/database"
+	"github.com/skynetlabs/pinner/skyd"
+	"gitlab.com/NebulousLabs/errors"
+	"gitlab.com/NebulousLabs/threadgroup"
+	"gitlab.com/SkynetLabs/skyd/build"
+)
+
+/**
+ PHASE 1: <DONE>
+ - scan the DB once a day for underpinned files
+ - when you find an underpinned file that's not currently locked
+	- lock it
+	- pin it locally and add the current server to its list
+	- unlock it
+
+ PHASE 2:
+ - calculate server load by getting the total number and size of files pinned by each server
+ - only pin underpinned files if the current server is in the lowest 20% of servers, otherwise exit before scanning further
+
+ PHASE 3:
+ - add a second scanner which looks for skylinks which should be unpinned and unpins them from the local skyd.
+*/
+
+var (
+	// SleepBetweenPins defines how long we'll sleep between pinning files.
+	// We want to add this sleep in order to prevent a single server from
+	// grabbing all underpinned files and overloading itself. We also want to
+	// allow for some time for the newly pinned files to reach full redundancy
+	// before we pin more files.
+	SleepBetweenPins = build.Select(
```

Comment:
The pin request returns when reaching redundancy 1 and skyd continues to work on that file until it reaches full redundancy.

Reply:
I don't get the effect a 10-second sleep has. It's completely arbitrary right? I might make no difference, it might make some difference? Is it just to help out a bit and what negative event happens if you would remove the sleep all together?

Reply:
From offline:

The main issue isn't the sleep but the fact that it's arbitrary. We decided to go with a semi-intelligent sleep which makes some sensible assumptions based on the size of the skylink - we assume a gigabit connection and using only 25% of that and standard redundancy.

Additional notes:
 - we don't care if this scanning process takes a long time, as long as it doesn't disrupt the normal operation of the server. scans cannot overlap, so even if it takes 3 days it's still fine.
 - the main purpose of the sleep is to protect against overloading the renter's repair queue
 - the secondary purpose of the sleep id to give other servers an opportunity to step in and pin some files, thus spreading the load

Reply:
So is the plan to handle updating the sleep in a follow up?

Reply:
Oh I see the implementation now. So is this variable now unused? Can we delete it?

Reply:
It is still being used as the default sleep time in case we fail to fetch the size of the skyfile from `skyd`.

I don't particularly like this approach, so if someone has a suggestion for improving https://github.com/SkynetLabs/pinner/pull/7/files#diff-febdfba77f54946ad0c77a437167750d33fe4aa988fdd0c8935f2fd90cb84442R216, I'd be happy to have it. After we remove it from there, we can remove it completely.

Reply:
https://github.com/SkynetLabs/pinner/issues/13